includes/class-windows-azure-file-contents-provider.php

```diff
@@ -146,8 +146,7 @@ public function __construct( $file_path, $chunk_size = self::CHUNK_SIZE ) {
 			$chunk_size = self::CHUNK_SIZE;
 		}
 		if ( '/' !== $file_path[0] && ( isset( $file_path[1] ) && ':' !== $file_path[1] ) ) {
-			$upload_dir = wp_upload_dir();
-			$file_path  = trailingslashit( $upload_dir['basedir'] ) . $file_path;
+			$file_path  = trailingslashit( \Windows_Azure_Helper::get_uploads_basedir() ) . $file_path;
```

Comment:
@ravichandrach1, `trailingslashit` is intended to add trailing slashes to URLs and shouldn't be used for file paths. Please, do not use it for your changes. Better use DIRECTORY_SEPARATOR constant in the `get_uploads_basedir` function to make sure we have directory separator at the end of basedir path.

Reply:
@eugene-manuilov  WordPress codex says the `trailingslashit` function is mainly for `paths` - https://codex.wordpress.org/Function_Reference/trailingslashit, should we not be using it ? I will move the `trailingslashit` to the `get_uploads_basedir` function and remove it from function invocations.